src/guards/user-associated-attendance.guard.ts

```diff
@@ -0,0 +1,25 @@
+import { CanActivate, ExecutionContext, Injectable } from '@nestjs/common';
+import { TimeSlotRepository } from '../repositories/time-slot.repository';
+import { GroupRepository } from '../repositories/group.repository';
+import { AttendanceRepository } from '../repositories/attendance.repository';
+import mongoose from 'mongoose';
+
+@Injectable()
+export class UserAssociatedAttendanceGuard implements CanActivate {
+  constructor(private readonly attendanceRepository: AttendanceRepository) {}
+
+  async canActivate(context: ExecutionContext): Promise<boolean> {
+    const request = context.switchToHttp().getRequest();
+
+    const attendance = await this.attendanceRepository.findOneById(
+      request.params.attendanceId,
+    );
+
+    if (!attendance) {
+      return false;
+    }
+
+    // @ts-ignore
```

Comment:
is it necessary ?

Reply:
Yes, because of TS which is screwed up with the custom template types in the template property.